src/VerifyEmailHelper.php

```diff
@@ -52,6 +57,9 @@ public function generateSignature(string $routeName, string $userId, string $use
 
         $extraParams['token'] = $this->tokenGenerator->createToken($userId, $userEmail);
         $extraParams['expires'] = $expiryTimestamp;
+        if (true === $this->useEmail) {
+            $extraParams['useremail'] = $userEmail;
```

Comment:
Ok, so the original reason that we didn't include the email in the URL was to avoid PII from being leaked into access log files. This was due to an abundance of caution, but it IS something that you will find done in other verification libraries.

So, if we do this, we should hash this email address - e.g. `sha1($userEmail)`. I might also:

A) call the parameter `hash`.
B) Rename the property in this class to `addEmailHash`
C) Rename the config option to `add_email_hash`

This will also require a bit more rework:

1) In the `$this->tokenGenerator->createToken()`, we would use the `sha1($userEmail)` instead of just `$userEmail` if the user has selected that option. Because we're hashing the email, we're going to use the hashed email during verification also, which means that the token we create needs to use the hashed email.

2) Below in `validateEmailConfirmation()`, we should probably rename `$userEmail` to `$userEmailOrHash` to clarify this could be the plain-text email or the email hash, depending on your "mode". We'll probably want some phpdoc above the method to clarify what's going on.

Reply:
By using a hashed email in the url there is also the need of storing that hash in the database to get the relation to the user. Is that ok with you? Using the plain email doesn't need that. I have to take a look first, where and how to store the hash. Nevertheless, I agree with the proposed changes.

Reply:
@jrushlow I am not sure if this is possible. As far as I know, you cannot hide a property (like a macro in c++) of an entity depending of an configuration date. (correct me if I'm wrong)